src/core/agent/services/identifierService.ts

```diff
@@ -85,6 +91,10 @@ class IdentifierService extends AgentService {
       const aid = await this.agent.modules.signify.getIdentifierByName(
         metadata.signifyName as string
       );
+      //Update multisig's status if it is pending
+      if (metadata.isPending && metadata.opName) {
+        await this.updateMultisigMetadata(metadata);
```

Comment:
Hmm, based on the contents of `updateMultisigMetadata`, we are just checking here if the `op` is done and then updating? Can we do this in an event driven way via the notifications to notify us when its down?

Reply:
Since the `op`'s status does not relate to the notification. We still need to check for the operation status. I think we can add an interval when the user receives the notification and then update the multisig. However, there is still a case like this: the user has accepted the join request but the operation takes too much time to be done, should we consider those notifications as unhandled and let users accept them again? Please let me know your thoughs about this case.

Reply:
OK, so from a UI perspective - this will always show pending in the list of identifiers, but once you click into it it will update to done if it's complete?

For me this is OK for this PR but we will need a follow up story to handle this better like you suggest, but happy to keep them as separate PRs :)

Reply:
`OK, so from a UI perspective - this will always show pending in the list of identifiers, but once you click into it it will update to done if it's complete?` 
Correct :D